storage/storage-proto/src/lib.rs

```diff
@@ -358,38 +360,117 @@ impl From<TreeState> for crate::proto::storage::TreeState {
 
 /// Helper to construct and parse [`proto::storage::StartupInfo`]
 #[derive(Clone, Debug, Eq, PartialEq)]
-#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
 pub struct StartupInfo {
+    /// The latest ledger info.
     pub latest_ledger_info: LedgerInfoWithSignatures,
-    pub ledger_info_with_validators: LedgerInfoWithSignatures,
+    /// If the above ledger info doesn't carry a validator set, the latest validator set. Otherwise
+    /// `None`.
+    pub latest_validator_set: Option<ValidatorSet>,
     pub committed_tree_state: TreeState,
     pub synced_tree_state: Option<TreeState>,
 }
 
+impl StartupInfo {
+    pub fn new(
+        latest_ledger_info: LedgerInfoWithSignatures,
+        latest_validator_set: Option<ValidatorSet>,
+        committed_tree_state: TreeState,
+        synced_tree_state: Option<TreeState>,
+    ) -> Self {
+        Self {
+            latest_ledger_info,
+            latest_validator_set,
+            committed_tree_state,
+            synced_tree_state,
+        }
+    }
+
+    pub fn get_validator_set(&self) -> Option<&ValidatorSet> {
```

Comment:
But before this change the callers are all using `expect`...

Reply:
that's because we can't type annotate those LedgerInfo must have validators.
if we have helper functions, we should enforce it inside to provide better API